filecoin-virtual-machine/fvm-contract-overview.md

```diff
@@ -6,12 +6,112 @@ description: >-
+
+If you are running a base interface (specifically, the one in the [RaaS Starter Kit](https://github.com/filecoin-project/raas-starter-kit)), there's an event listener inside the RaaS service node that you can use to listen for new deal requests.
+This event listener performs processing for each job submitted to the contract to add RaaS service workers and eventually to call `complete` on the contract.
+
+```javascript
+// Initialize the listener for the Deal Creation event
+async function initializeDealCreationListener() {
+  const dealStatus = await ethers.getContractAt(contractName, contractInstance);
+
+  /// Logic for handling SubmitAggregatorRequest events
+  function handleEvent(transactionId, cid) {
+    console.log(`Received SubmitAggregatorRequest event: (Transaction ID: ${transactionId}, CID: ${cid})`);
+    // ... other code to handle the event emission
+
+    (async () => {
+      // ... other code
+      
+      // After processing this event, reattach the event listener
+      if (dealStatus.listenerCount("SubmitAggregatorRequest") === 0) {
+        dealStatus.once("SubmitAggregatorRequest", handleEvent);
+      }
+    })();
+  }
+
+  // Start listening to the first event and recursively handle the next events
+  if (dealStatus.listenerCount("SubmitAggregatorRequest") === 0) {
+    dealStatus.once("SubmitAggregatorRequest", handleEvent);
+  }
+}
+```
+
+To use this, simply do `yarn service` in the terminal and proceed through the frontend as you normally would - uploading any random file, and then registering the workers using the autocompleted CID that appears in the box. **If you want to register the workers manually for a job that you didn't upload, just simply paste in the known CID of your file into the box and register the jobs anyway**.
+
+With Lighthouse's full interface, you'll have to make a few additions instead to the API call on upload. Recall the process described for uploading a file with replication:
+
+```javascript
+import lighthouse from "@lighthouse-web3/sdk";
+// ... other code
+// Indicates that the deal to maintain your file will be replicated to a total of two copies on the network.
+const dealParams = {
+  num_copies: 2,
+};
+// The `false` indicates that we're uploading a single file.
+// Returns a CID (Content ID) for your file that you can use for PoDSI verification.
+const uploadResponse = await lighthouse.upload('/path/to/adorable/dog.jpg', 'YOUR_API_KEY', false, dealParams);
+```
+
+For initializing RaaS services via. the full interface, you'll need to add a few more parameters to the `dealParams` object:
+
+```javascript
+import lighthouse from "@lighthouse-web3/sdk";
+// ... other code
+
+// Deal params should include the settings for RaaS jobs
+const dealParams = {
+  num_copies: 2,
+  repair_threshold: 28800,
+  renew_threshold: 240,
+};
+
+const uploadResponse = await lighthouse.upload('/path/to/adorable/dog.jpg', 'YOUR_API_KEY', false, dealParams);
+```
```

Comment:
This also should reference https://calibration.lighthouse.storage/api/register_job and https://calibration.lighthouse.storage/api/deal_status?cid=QmYSNU2i62v4EFvLehikb4njRiBrcWqH6STpMwduDcNmK6

Reply:
Since these allow you to register jobs after upload.